Docs: Move and rename public dashboards page (#91817)
grafana/grafana: docs/sources/dashboards/share-dashboards-panels/shared-dashboards/index.md

  caching:
    - pattern: /docs/grafana/
      destination: /docs/grafana/<GRAFANA_VERSION>/administration/data-source-management/#query-and-resource-caching
    - pattern: /docs/grafana-cloud/
      destination: /docs/grafana/<GRAFANA_VERSION>/administration/data-source-management/#query-and-resource-caching
  grafana-enterprise:
    - pattern: /docs/grafana/
      destination: /docs/grafana/<GRAFANA_VERSION>/introduction/grafana-enterprise/
    - pattern: /docs/grafana-cloud/
      destination: /docs/grafana/<GRAFANA_VERSION>/introduction/grafana-enterprise/
---

# Externally shared dashboards

<!-- Update aliases -->

{{< admonition type="warning" >}}
Sharing your dashboard externally could result in a large number of queries to the data sources used by your dashboard.
This can be mitigated by using the Enterprise [caching](ref:caching) and/or rate limiting features.
{{< /admonition >}}

Externally shared dashboards allow you to share your Grafana dashboard with anyone. This is useful when you want to make your dashboard available to the world without requiring access to your Grafana organization.

> You can see a list of all your externally shared dashboards in one place by navigating to **Dashboards > Shared dashboards**. For each dashboard in the list, the page displays the status, a link to view the dashboard, a link to the externally shared dashboard configuration, and the options to pause or revoke the external URL.

Reviewer: There are 2 dashboard links.
The first one redirects you to the real dashboard and is when you click the whole card
The second one (I think that's the one you refer to) ,is to view the public dashboard. Is it ok to say "a link to view the dashboard"? because that's not true, currently the public dashboard is not exactly as the dashboard (template variables, unsupported data sources, etc).
Should we say "a link to view the shared/external dashboard"?

Author: I'm addressing this on #90630

## Security implications of sharing your dashboard externally

- Anyone with the URL can access the dashboard.
- Externally shared dashboards are read-only.
- Arbitrary queries **cannot** be run against your data sources through externally shared dashboards. Externally shared dashboards can only execute the
  queries stored on the original dashboard.

### Share externally to anyone with a link

To share your dashboard so that anyone with the link can access it, follow these steps.

1. Click **Dashboards** in the main menu.
1. Click the dashboard you want to share.
1. Click the **Share** drop-down in the top-right corner and select **Share externally**.
1. In the **Link access** drop-down, select **Anyone with the link**.
1. Click the checkbox confirming that you understand the entire dashboard will be public.
1. Click **Accept**.
1. (Optional) Set the following options:
   - **Enable time range** - Allow people accessing the link to change the time range. This configuration screen shows the default time range of the dashboard.
   - **Display annotations** - Allow people accessing the link to view the dashboard annotations.

> 1. Each recipient will receive an email with link to the dashboard that's valid until you revoke access to the dashboard.

Reviewer: this is not true. there's no email in "anyone with the link"

1. Click the **X** at the top-right corner to close the share drawer.

Once you've shared a dashboard externally, a **Public** label is displayed in the header of the dashboard.

#### Update access to an external dashboard link

You can update the access to externally shared dashboard links by following these steps:

1. Click **Dashboards** in the main menu.
1. Click the dashboard you want to share.
1. Click the **Share** drop-down in the top-right corner and select **Share externally**.
1. Do one of the following:
   - Click **Pause access** so that people can't access the dashboard, but the link is maintained.
   - Click **Resume access** so that people can access the dashboard again.
   - Click **Revoke access** so that people can't access the dashboard unless a new external link is generated. Confirm that you want to revoke the link.
1. Click the **X** at the top-right corner to close the share drawer.

> The link no longer works. You must create a new external URL, as in [Share externally to anyone with a link](#share-externally-to-anyone-with-a-link).

Reviewer 2: Should we keep it consistent and use `external link` over ` external URL`

## Share externally with specific people

{{< admonition type="note" >}}
Available in [Grafana Enterprise](ref:grafana-enterprise) and [Grafana Cloud](/docs/grafana-cloud).
{{< /admonition >}}

To share with specific external users, you can send them a link by email. Use this option when you only want to share your dashboard with specific people instead of anyone who navigates to the link. When you use email sharing, recipients receive a one-time use link that's valid for **one hour**. Once the link is used, the viewer has access to the shared dashboard for **30 days**.

When you share a dashboard with an email link, your organization is billed per user for the duration of the 30-day token, regardless of how many dashboards are shared. Billing stops after 30 days unless you renew the token.

1. Click **Dashboards** in the main menu.
1. Click the dashboard you want to share.
1. Click the **Share** drop-down in the top-right corner and select **Share externally**.

> 1. In the **Link access** drop-down, select **Anyone with the link**.

Reviewer: ```suggestion
1. In the **Link access** drop-down, select **Only specific people**.
```

1. Click the checkbox confirming that you understand payment is required to add users.
1. Click **Accept**.
1. In the **Invite** field, enter the email address of the person you want to invite and click **Invite**.

   You can only invite one person at a time.

1. (Optional) Set the following options:
   - **Enable time range** - Allow people accessing the link to change the time range. This configuration screen shows the default time range of the dashboard.
   - **Display annotations** - Allow people accessing the link to view the dashboard annotations.
1. Click **Copy external link**.
1. Send the copied URL to any external user.
1. Click the **X** at the top-right corner to close the share drawer.

### Viewers requesting access

If a viewer without access tries to navigate to the shared dashboard, they'll be asked to request access by providing their email. They'll receive an email with a new one-time use link if the email they provided has already been invited to view the shared dashboard and hasn't been revoked.

> If the viewer doesn't have an invitation or it's been revoked, you won't be notified and no link is sent.

Reviewer: it's true no link is sent.
But you will never be notified, no matter the viewer has an invitation or not. 
I mean, it's a silence process.
If the user **is** in the email list, when he enters his email, he will receive a link **but** the "admin" won't be notified

1. Click **Dashboards** in the main menu.
1. Click the dashboard you want to share.
1. Click the **Share** drop-down in the top-right corner and select **Share externally**.
1. Click the menu icon (three dots) next to the email address of the viewer for whom you'd like to revoke access.
1. Click **Revoke access**.

The viewer immediately no longer has access to the dashboard, nor can they use any existing one-time use links they may have.

### Re-invite a viewer

1. Click **Dashboards** in the main menu.
1. Click the dashboard you want to share.
1. Click the **Share** drop-down in the top-right corner and select **Share externally**.
1. Click the menu icon (three dots) next to the email address of the viewer you'd like to invite again.
1. Click **Resend invite**.

The viewer receives an email with a new one-time use link. This invalidates all previously issued links for that viewer.

### View shared dashboard users

To see a list of users who have accessed your externally shared dashboard by way of email sharing, take the following steps:

1. In the main sidebar navigation, click **Administration**.
1. Click **Users**.

> 1. Click the **Public dashboard users** tab.

Reviewer: ```suggestion
1. Click the **Shared dashboard users** tab.
```
